IdentityCore/src/util/NSString+MSIDExtensions.m

```diff
@@ -243,6 +243,11 @@ + (NSString *)msidEncodeStringFromDictionary:(NSDictionary *)dict
          {
              [encodedString appendFormat:@"=%@", encodedValue];
          }
+        
+         else if ((encodedValue && ![encodedValue isKindOfClass:[NSNull class]]) && !encodedValue.length)
```

Comment:
What is the difference to check ```!encodedValue``` vs ```![encodedValue isKindOfClass:[NSNull class]]```?
Is there a way to simplify the logic?

Reply:
It's for a very rare corner case - From Apple Doc: 

The NSNull class defines a singleton object you use to represent null values in situations where nil is prohibited as a value (typically in a collection object such as an array or a dictionary).

Reply:
@NerevarineRule, let's add a unit test for these cases?